<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>JDFOAM_GUI</class>
 <widget class="QMainWindow" name="JDFOAM_GUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>630</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>630</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>630</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>JDFOAM</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>5</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>10</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>5</number>
        </property>
        <item>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="case_label">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <hintingpreference>PreferNoHinting</hintingpreference>
             </font>
            </property>
            <property name="text">
             <string>算例目录:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="case_path_edit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">border-radius: 6px;</string>
            </property>
            <property name="placeholderText">
             <string>选择算例项目根目录</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="msh_label">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <hintingpreference>PreferNoHinting</hintingpreference>
             </font>
            </property>
            <property name="text">
             <string>.msh文件:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="msh_path_edit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>30</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">border-radius: 6px;</string>
            </property>
            <property name="placeholderText">
             <string>选择待转换的 .msh 文件</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <property name="horizontalSpacing">
           <number>5</number>
          </property>
          <item row="0" column="0">
           <widget class="QPushButton" name="case_browse_btn">
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="toolTip">
             <string>浏览文件夹</string>
            </property>
            <property name="styleSheet">
             <string notr="true">/* 按钮默认样式 */
QPushButton {
    color: white;
    border-radius: 6px;
    border: 1px solid rgb(160, 160, 160); 
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(160, 160, 160); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 3px;
}
</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset theme="folder-open"/>
            </property>
            <property name="iconSize">
             <size>
              <width>25</width>
              <height>25</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="case_open_btn">
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="toolTip">
             <string>打开文件夹</string>
            </property>
            <property name="styleSheet">
             <string notr="true">/* 按钮默认样式 */
QPushButton {
    color: white;
    border-radius: 6px;
    border: 1px solid rgb(160, 160, 160); 
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(160, 160, 160); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 3px;
}
</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset theme="edit-find"/>
            </property>
            <property name="iconSize">
             <size>
              <width>25</width>
              <height>25</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="treefoam_btn">
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="toolTip">
             <string>启动TreeFoam</string>
            </property>
            <property name="styleSheet">
             <string notr="true">/* 按钮默认样式 */
QPushButton {
    color: white;
    border-radius: 6px;
    border: 1px solid rgb(160, 160, 160); 
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(160, 160, 160); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 3px;
}
</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>TreeFoam.png</normaloff>TreeFoam.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>25</width>
              <height>25</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="msh_browse_btn">
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="toolTip">
             <string>浏览.msh文件路径</string>
            </property>
            <property name="styleSheet">
             <string notr="true">/* 按钮默认样式 */
QPushButton {
    color: white;
    border-radius: 6px;
    border: 1px solid rgb(160, 160, 160); 
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(160, 160, 160); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 3px;
}
</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset theme="folder-open"/>
            </property>
            <property name="iconSize">
             <size>
              <width>25</width>
              <height>25</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="msh_open_btn">
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="toolTip">
             <string>打开.msh所在文件夹</string>
            </property>
            <property name="styleSheet">
             <string notr="true">/* 按钮默认样式 */
QPushButton {
    color: white;
    border-radius: 6px;
    border: 1px solid rgb(160, 160, 160); 
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(160, 160, 160); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 3px;
}
</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset theme="edit-find"/>
            </property>
            <property name="iconSize">
             <size>
              <width>25</width>
              <height>25</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="gmsh_btn">
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="toolTip">
             <string>启动Gmsh</string>
            </property>
            <property name="styleSheet">
             <string notr="true">/* 按钮默认样式 */
QPushButton {
    color: white;
    border-radius: 6px;
    border: 1px solid rgb(160, 160, 160); 
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(160, 160, 160); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 3px;
}
</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>gmsh.ico</normaloff>gmsh.ico</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>25</width>
              <height>25</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="button_hbox">
        <property name="spacing">
         <number>20</number>
        </property>
        <item>
         <widget class="QPushButton" name="start_mesh_btn">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <hintingpreference>PreferNoHinting</hintingpreference>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">/* 按钮默认样式 */
QPushButton {
    background-color:rgb(167, 83, 250);
    color: white;
    border-radius: 6px;
    padding: 5px 15px;
    border: none;
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color:rgb(127, 64, 191);
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 6px;
}
</string>
          </property>
          <property name="text">
           <string>开始转换网格</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="combine_md_btn">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <hintingpreference>PreferNoHinting</hintingpreference>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">/* 按钮默认样式 */
QPushButton {
    background-color: rgb(63, 140, 243);
    color: white;
    border-radius: 6px;
    padding: 5px 15px;
    border: none;
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(53, 118, 204); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 6px;
}
</string>
          </property>
          <property name="text">
           <string>合并代码为Markdown</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="combine_pdf_btn">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>40</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <hintingpreference>PreferNoHinting</hintingpreference>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">/* 按钮默认样式 */
QPushButton {
    background-color:rgb(255, 160, 162);
    border-radius: 6px;
    padding: 5px 15px;
    border: none;
}

/* 悬浮特效：背景颜色变浅，并增加蓝色边框感 */
QPushButton:hover {
    background-color: rgb(0, 200, 0); /* 颜色比默认稍亮 */
}

/* 按下特效：点击时颜色变深，产生物理压下的错觉 */
QPushButton:pressed {
    padding-top: 6px;
}
</string>
          </property>
          <property name="text">
           <string>导出代码为PDF</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QProgressBar" name="progress_bar">
        <property name="styleSheet">
         <string notr="true">/* 进度条背景槽 */
QProgressBar {
    border: none;                /* 去掉边框 */
    color: white;                /* 百分比文字颜色 */
    text-align: center;          /* 文字居中 */
    background: #e0e0e0;         /* 槽的背景色 */
    border-radius: 5px;          /* 圆角高度的一半通常看起来很舒服 */
    height: 10px;                /* 这里的 height 会影响槽的粗细 */
}

/* 已填充的进度部分 */
QProgressBar::chunk {
    background-color: #05B8CC;   /* 进度条颜色：青蓝色 */
    border-radius: 5px;          /* 进度条的圆角 */
}</string>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="textVisible">
         <bool>false</bool>
        </property>
        <property name="format">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="Log">
        <property name="styleSheet">
         <string notr="true">QPlainTextEdit {
    /* 1. 必须有边框，透明即可 */
    border: 1px solid transparent;
    
    /* 2. 定义圆角 */
    border-radius: 6px;
    
    /* 3. 核心：必须定义背景色，圆角才能被“填充”出来 */
    /* palette(base) 会自动跟随主题：浅色时是白色，深色时是深灰 */
    background-color: palette(base);
    
    /* 4. 文字颜色也跟随主题 */
    color: palette(text);
    
    /* 建议加点内边距，否则字会贴到圆角边上 */
    padding: 3px;
}</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>630</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_theme">
    <property name="title">
     <string>Theme</string>
    </property>
    <addaction name="action_light_theme"/>
    <addaction name="action_dark_theme"/>
   </widget>
   <addaction name="menu_theme"/>
  </widget>
  <action name="action_light_theme">
   <property name="text">
    <string>light</string>
   </property>
   <property name="font">
    <font>
     <hintingpreference>PreferNoHinting</hintingpreference>
    </font>
   </property>
  </action>
  <action name="action_dark_theme">
   <property name="text">
    <string>dark</string>
   </property>
   <property name="font">
    <font>
     <hintingpreference>PreferNoHinting</hintingpreference>
    </font>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
